### ISSUE TITLE
Support for macOS

### ISSUE THREAD

**Issue by scizzorz** [scizzorz/rain#123]

**Comment by scizzorz** [comment]
Or maybe it does. I don't know. I don't have a Mac. According to @tubebaum, the `_Unwind_*` methods can be found in `libgcc_s.1.dylib` rather than `libgcc_s.so.1` on most Linux platforms.

**Comment by checkthebounds** [comment]
it's `libunwind.dylib`

**Comment by scizzorz** [comment]
It seems like `libunwind` probably comes from XCode, but `GC_init()` needs to be called on macOS, which is optional on Linux.

The `:libgcc_s.so.1` library will need to be changed in `core/_pkg.rn` to work cross-platform as well. I think it's only needed on Arch Linux anyway - for some reason the `_Unwind_*` function definitions don't live in `libunwind`...

**Comment by scizzorz** [comment]
Someone sent me a git diff of the necessary changes to run on macOS, so I'll clean this up eventually:

```
diff --git a/core/_pkg.rn b/core/_pkg.rn
index 224c4c3..4f810ed 100644
--- a/core/_pkg.rn
+++ b/core/_pkg.rn
@@ -8,7 +8,7 @@ link "table/table.c"
 
 library "gc"
 library "unwind"
-library ":libgcc_s.so.1"
+library "gcc_s.1"
 
 import env
 import except
diff --git a/core/env/env.c b/core/env/env.c
index 87c54fa..be8215c 100644
--- a/core/env/env.c
+++ b/core/env/env.c
@@ -1,9 +1,10 @@
 #include "../rain.h"
 #include <stdlib.h>
 #include <string.h>
-
+#include <gc.h>
 
 void rain_init_args(int argc, char **argv) {
+  GC_init();
   box key;
   box val;
   for(int i = 0; i < argc; ++i) {
diff --git a/core/set/set.c b/core/set/set.c
index 9fe2858..649e599 100644
--- a/core/set/set.c
+++ b/core/set/set.c
@@ -3,6 +3,10 @@
 #include <stdlib.h>
 #include <string.h>
 
+int rain_gcinit() {
+    GC_init();
+    return 0;
+}
 
 // box helpers
 
diff --git a/rain/parser.py b/rain/parser.py
index 5787167..b43fcf6 100644
--- a/rain/parser.py
+++ b/rain/parser.py
@@ -5,6 +5,7 @@ from . import error as Q
 from . import module as M
 from . import token as K
 from . import types as T
+import ctypes
 from ctypes import byref
 from os import environ as ENV
 from os.path import join
@@ -77,6 +78,9 @@ class macro:
     self.eng.link_file(ast.ll, *ast.links)
     self.eng.add_lib(so)
     self.eng.finalize()
+    gcinit = self.eng.get_func('rain_gcinit', ctypes.c_int)
+    if gcinit:
+      gcinit()
 
   def parse(self, ctx):
     return [fn(ctx) for fn in self.parses]
```

**Comment by scizzorz** [comment]
The `GC_init()` is implemented in 4cd5cda505501d32923c50536f48a598fc95a484
